# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/VideoDecoder.cpp
# Commit: f19cfef38d63
# Full Hash: f19cfef38d63df3cd02b0413e2eea2f81075e649
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-29 15:31:45
# Regressor Bug: 1831451
# File Overlap Count: 1
# Description:
#   Bug 1831451 - Allow constructing VideoFrame without pixel format r=padenot
#   
#   Depends on D171166
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171518
# ==============================================================================

diff -r 1cf50ce36a73 -r f19cfef38d63 dom/media/webcodecs/VideoDecoder.cpp
--- a/dom/media/webcodecs/VideoDecoder.cpp	Thu Jun 29 04:15:03 2023 +0000
+++ b/dom/media/webcodecs/VideoDecoder.cpp	Thu Jun 29 04:15:03 2023 +0000
@@ -535,12 +535,10 @@
     // TODO: Calculate displaySize
   }
 
-  // TODO: Allow constructing VideoFrame without pixel format
-  return MakeRefPtr<VideoFrame>(
-      aGlobalObject, aData->mImage,
-      format ? format.ref() : VideoPixelFormat::EndGuard_,
-      aData->mImage->GetSize(), aData->mImage->GetPictureRect(), displaySize,
-      Some(aDuration), aTimestamp, aColorSpace);
+  return MakeRefPtr<VideoFrame>(aGlobalObject, aData->mImage, format,
+                                aData->mImage->GetSize(),
+                                aData->mImage->GetPictureRect(), displaySize,
+                                Some(aDuration), aTimestamp, aColorSpace);
 }
 
 static nsTArray<RefPtr<VideoFrame>> DecodedDataToVideoFrames(